sdk/core/azure-core/inc/azure/core/http/curl_transport.hpp

```diff
@@ -151,6 +178,15 @@ namespace Azure { namespace Core { namespace Http {
      */
     bool HttpKeepAlive = true;
 
+    /**
+     * @brief Options specified in the keep-alive request header
+     *
+     * @remark The keep-alive feature allows the SDK to reuse the same connection to the service for
+     * multiple requests. This field is populated if the Keep-Alive header is present in the
+     * request.
+     */
+    Azure::Nullable<Azure::Core::Http::_detail::KeepAliveOptions> KeepAliveOptions;
```

Comment:
QQ: Why expose these as options to customers? Do we believe customers will want to tweak the defaults?

I have no opinions here, I'm just curious about the logic of adding it to the CurlTransportOptions rather than keeping it an internal construct in the connection pool.

Reply:
since this is a header on the request i thought the data should be on the request object

Reply:
It's a header on the request AND the response. But I thought this change was on the CurlTransportOptions, which means it should probably apply to ALL requests.

I'm just curious about that part of the design.

Reply:
I put it there because that's where the HttpKeepAlive field is located, if HttpKeepAlive ison the transport options i suspect that's where the keepalive options should also be